client/src/modules/purchases/create/createUpdate.js

```diff
@@ -164,26 +170,36 @@ function PurchaseOrderController(Purchases, PurchaseOrder, Notify, Session, util
 
   function optimalPurchase() {
     vm.optimalPO = true;
+    const filters = {
+      includeEmptyLot : 0,
+      period : 'allTime',
+      require_po : 1,
+    };
 
-    Stock.inventories.read(null, { require_po : 1 })
+    Stock.inventories.read(null, filters)
       .then(rows => {
         if (!rows.length) {
           return Notify.warn('FORM.INFO.NO_INVENTORY_PO');
         }
 
+        const optimalPurchaseData = Purchases.formatOptimalPurchase(vm.inventories, rows);
+
         // adding items.length line in the Order store, which will be reflected to the grid
-        if (rows.length > 1) {
-          vm.order.addItem(rows.length);
+        if (optimalPurchaseData > 1) {
+          vm.order.addItem(optimalPurchaseData.length);
         }
 
         vm.order.store.data.forEach((item, index) => {
-          item.code = rows[index].code;
-          item.inventory_uuid = rows[index].inventory_uuid;
-          item.description = rows[index].text;
-          item.quantity = rows[index].S_Q;
-          item.unit_price = 0;
-          item.unit = rows[index].unit_type;
-          item._initialised = true;
+          item.inventory_uuid = optimalPurchaseData[index].uuid;
```

Comment:
It looks like, if there is data in the grid, it will replace the old data with new data.

I propose that you just go ahead and clear the grid with `vm.order.clear()` and then add the appropriate number of elements to the grid.